src/main/java/com/gluonhq/substrate/target/AbstractTargetConfiguration.java

```diff
@@ -287,7 +287,14 @@ public String run(Path appPath, String appName) throws IOException, InterruptedE
      */
     @Override
     public boolean runUntilEnd() throws IOException, InterruptedException {
-        Process runProcess = startAppProcess(paths.getAppPath(), projectConfiguration.getAppName());
+        Path appPath = Objects.requireNonNull(paths.getAppPath(),
+                "Application path can't be null");
+        Path app = appPath.resolve(Objects.requireNonNull(projectConfiguration.getAppName(),
+                "Application name can't be null"));
```

Comment:
I particularly don't like this line: checking for `null` inside a method call. May be its just me :)

Reply:
Fixed